internal/ingress/election/election.go

```diff
@@ -35,9 +36,31 @@ type elector struct {
 	elector *leaderelection.LeaderElector
 }
 
-func (e elector) Run() {
-	// TODO pass the correct context in here
-	e.elector.Run(context.Background())
+func (e elector) Run(ctx context.Context) {
+	backoff := flowcontrol.NewBackOff(1*time.Second, 15*time.Second)
+
+	for session := 0; ; {
```

Comment:
Let's rename this to `retryCount` and define it in the method itself and change the for loop to be infinite:

```go
for {
}
```